OWML.ModHelper/ModConfig.cs

```diff
@@ -104,27 +108,82 @@ public void MakeConsistentWithDefaults(IModConfig defaultConfig)
                 return;
             }
 
-            AddMissingDefaults(defaultConfig);
+            var toRemove = Settings.Keys.Except(defaultConfig.Settings.Keys).ToList();
+            toRemove.ForEach(key => Settings.Remove(key));
 
-            var settingsCopy = new Dictionary<string, object>(Settings);
-            foreach (var setting in Settings)
+            var keysCopy = Settings.Keys.ToList();
+            foreach (var key in keysCopy)
             {
-                if (!defaultConfig.Settings.ContainsKey(setting.Key))
+                if (!IsSettingSameType(Settings[key], defaultConfig.Settings[key]))
                 {
-                    settingsCopy.Remove(setting.Key);
+                    TryUpdate(key, Settings[key], defaultConfig.Settings[key]);
                 }
-                else if (!IsSettingSameType(setting.Value, defaultConfig.Settings[setting.Key]))
+                else if (defaultConfig.Settings[key] is JObject objectValue && objectValue["type"].ToString() == "selector")
                 {
-                    settingsCopy[setting.Key] = defaultConfig.Settings[setting.Key];
+                    UpdateSelector(key, Settings[key], objectValue);
                 }
             }
-            Settings = settingsCopy;
+
+            AddMissingDefaults(defaultConfig);
+        }
+
+        private bool UpdateSelector(string key, object userSetting, JObject modderSetting)
+        {
+            var options = modderSetting["options"].ToObject<List<string>>();
+            var userString = userSetting is JObject objectValue ? (string)objectValue["value"] : Convert.ToString(userSetting);
+            Settings[key] = modderSetting;
+            if (options.Contains(userString))
+            {
+                SetSettingsValue(key, userString);
+                return true;
+            }
+            return false;
         }
 
         private void AddMissingDefaults(IModConfig defaultConfig)
         {
             var missingSettings = defaultConfig.Settings.Where(s => !Settings.ContainsKey(s.Key)).ToList();
             missingSettings.ForEach(setting => Settings.Add(setting.Key, setting.Value));
+        }   
+
+        private bool TryUpdate(string key, object userSetting, object modderSetting)
```

Comment:
this method is very long.
if you return in the ifs instead of changing state, it would be easier to break it up into smaller methods

Reply:
I don't think it's possible to return in ifs

Reply:
I feel like moving `Settings[key] = modderSetting;` to start might affect userSettings, imma test that

Reply:
alright, doesn't seem to be the case. Shortened TryUpdate a little bit